### PR TITLE
services/horizon/docker: Remove NETWORK_MODE=host configuration for linux users

### PR DESCRIPTION
**Pull request by tamirms**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Fixes https://github.com/stellar/go/issues/3764

Docker for linux recently implemented a fix which allows a container to access the host machine by setting the `extra_hosts` property https://stackoverflow.com/questions/31324981/how-to-access-host-port-from-docker-container/61424570#61424570

With that fix we can get rid of the `network_mode=host` configuration

### CODE REVIEW

**Comment by Shaptic** [comment]
**TL;DR:** This works, but is insufficient; we also need #3769. Thank you @tamirms!

With this fix, I can confirm that the error from #3764 goes away on Arch Linux. I can also confirm the following:

 - By itself, it isn't sufficient in getting all 4 containers _running_ in `standalone` mode.
 - However, when used alongside #3769, then all 4 _do_ run.
 - In `testnet` mode, it _is_ sufficient for getting the 2 containers going.

Under `standalone` mode, we have a new issue to overcome: the Horizon ingestion state machine fails to start.

```
Error in ingestion state machine

current_state="buildFromCheckpoint(checkpointLedger=0)" 
error="unexpected checkpointLedger value" 
next_state=start pid=1 service=ingest
```

However, this warrants a separate issue which I'll open.

**Review by Shaptic — APPROVED** [review]
